Index: ocaml-libvirt/libvirt/Makefile.in
===================================================================
--- ocaml-libvirt.orig/libvirt/Makefile.in	2008-06-28 18:50:51.000000000 +0200
+++ ocaml-libvirt/libvirt/Makefile.in	2008-06-28 18:55:32.000000000 +0200
@@ -124,8 +124,15 @@
 libvirt_version.cmo: libvirt_version.cmi
 libvirt_version.cmi: libvirt_version.mli
 
-install:
-	ocamlfind install libvirt \
-	  ../META *.so *.a *.cmx *.cma *.cmxa *.cmi *.mli
+
+install-byte:
+	mkdir -p $(OCAMLDESTDIR)
+	ocamlfind install -destdir $(OCAMLDESTDIR) -ldconf ignore libvirt \
+	  ../META *.so *.a *.cma *.cmi *.mli
+
+install-opt:
+	mkdir -p $(OCAMLDESTDIR)
+	ocamlfind install -destdir $(OCAMLDESTDIR) -ldconf ignore libvirt \
+	  ../META *.so *.a *.cma *.cmx *.cmxa *.cmi *.mli
 
 include ../Make.rules
Index: ocaml-libvirt/Makefile.in
===================================================================
--- ocaml-libvirt.orig/Makefile.in	2008-06-28 18:50:51.000000000 +0200
+++ ocaml-libvirt/Makefile.in	2008-06-28 18:55:32.000000000 +0200
@@ -27,7 +27,7 @@
 
 SUBDIRS		= libvirt mlvirsh examples
 
-all opt depend install:
+all opt depend install-opt install-byte:
 	for d in $(SUBDIRS); do \
 	  $(MAKE) -C $$d $@; \
 	  if [ $$? -ne 0 ]; then exit 1; fi; \
@@ -141,4 +141,4 @@
 force:
 
 .PHONY: all opt depend install clean distclean configure dist check-manifest \
-	release release_stage_2 release_stage_3 force
\ No newline at end of file
+	release release_stage_2 release_stage_3 force
Index: ocaml-libvirt/examples/Makefile.in
===================================================================
--- ocaml-libvirt.orig/examples/Makefile.in	2008-06-28 18:50:51.000000000 +0200
+++ ocaml-libvirt/examples/Makefile.in	2008-06-28 18:55:32.000000000 +0200
@@ -54,6 +54,6 @@
 	  $(OCAMLOPTPACKAGES) $(OCAMLOPTFLAGS) $(OCAMLOPTLIBS) \
 	  ../libvirt/mllibvirt.cmxa -o $@ $<
 
-install:
+install-opt install-byte:
 
 include ../Make.rules
Index: ocaml-libvirt/mlvirsh/Makefile.in
===================================================================
--- ocaml-libvirt.orig/mlvirsh/Makefile.in	2008-06-28 18:50:51.000000000 +0200
+++ ocaml-libvirt/mlvirsh/Makefile.in	2008-06-28 18:56:17.000000000 +0200
@@ -84,10 +84,16 @@
 	  ../libvirt/mllibvirt.cmxa -o $@ $^
 endif
 
-install:
+install-opt:
 	if [ -x mlvirsh.opt ]; then \
 	  mkdir -p $(DESTDIR)$(bindir); \
 	  $(INSTALL) -m 0755 mlvirsh.opt $(DESTDIR)$(bindir)/mlvirsh; \
 	fi
 
+install-byte:
+	if [ -x mlvirsh ]; then \
+	  mkdir -p $(DESTDIR)$(bindir); \
+	  $(INSTALL) -m 0755 mlvirsh $(DESTDIR)$(bindir)/mlvirsh; \
+	fi
+
 include ../Make.rules
